src/test/csfle.rs

```diff
@@ -0,0 +1,1261 @@
+            .key_vault_namespace(KV_NAMESPACE.clone())
+            .key_vault_client(kv_client.unwrap_or_else(|| client.into_client()))
+            .kms_providers(LOCAL_KMS.clone())
+            .build();
+        let client_encryption = ClientEncryption::new(enc_opts)?;
+
+        // Test: encrypted client.
+        let result = client_encrypted
+            .database("db")
+            .collection::<Document>("coll")
+            .insert_one(doc! { "encrypted": "test" }, None)
+            .await;
+        if with_external_key_vault {
+            assert!(result.is_err());
+        } else {
+            assert!(
+                result.is_ok(),
+                "unexpected error: {}",
+                result.err().unwrap()
+            );
+        }
+        // Test: manual encryption.
+        let result = client_encryption
+            .encrypt(
+                RawBson::String("test".to_string()),
+                &EncryptOptions::builder()
+                    .key(EncryptKey::Id(base64_uuid("LOCALAAAAAAAAAAAAAAAAA==")?))
+                    .algorithm(Algorithm::AeadAes256CbcHmacSha512Deterministic)
+                    .build(),
+            )
+            .await;
+        if with_external_key_vault {
+            assert!(result.is_err());
+        } else {
+            assert!(
+                result.is_ok(),
+                "unexpected error: {}",
+                result.err().unwrap()
+            );
+        }
+    }
+
+    Ok(())
+}
+
+fn load_testdata_raw(name: &str) -> Result<String> {
+    let path: PathBuf = [
+        env!("CARGO_MANIFEST_DIR"),
+        "src/test/spec/json/client-side-encryption/testdata",
+        name,
+    ]
+    .iter()
+    .collect();
+    Ok(std::fs::read_to_string(path)?)
+}
+
+fn load_testdata(name: &str) -> Result<Document> {
+    Ok(serde_json::from_str(&load_testdata_raw(name)?)?)
+}
+
+// Prose test 4. BSON Size Limits and Batch Splitting
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+async fn bson_size_limits() -> Result<()> {
+    if !check_env("bson_size_limits", false) {
+        return Ok(());
+    }
+    let _guard = LOCK.run_exclusively().await;
+
+    // Setup: db initialization.
+    let (client, datakeys) = init_client().await?;
+    client
+        .database("db")
+        .create_collection(
+            "coll",
+            CreateCollectionOptions::builder()
+                .validator(doc! { "$jsonSchema": load_testdata("limits-schema.json")? })
+                .build(),
+        )
+        .await?;
+    datakeys
+        .insert_one(load_testdata("limits-key.json")?, None)
+        .await?;
+
+    // Setup: encrypted client.
+    let mut opts = CLIENT_OPTIONS.get().await.clone();
+    let handler = Arc::new(EventHandler::new());
+    let mut events = handler.subscribe();
+    opts.command_event_handler = Some(handler.clone());
+    // TODO: register an event handler
```

Comment:
it looks like we register a handler just before this. should this TODO still be here?